api/advanced-topics/python-extension-template.md

```diff
@@ -0,0 +1,82 @@
+
+The TypeScript part handles working with VS Code and its UI. The extension template comes with a few settings built-in that can be used by your tool. If you need to add new settings to support your tool, you will have to work with a bit of TypeScript. The extension template has examples for a few settings and you can also look at [extensions developed](#examples) by our team for some of the popular tools.
+
+## Requirements
+
+1. VS Code 1.64.0 or greater
+1. Python 3.7 or greater
+1. node >= 14.19.0
+1. npm >= 8.3.0 (`npm` is installed with node, check npm version, use `npm install -g npm@8.3.0` to update)
+1. Python extension for VS Code
+
+You should know to create and work with Python virtual environments.
+
+## Getting started
+
+To get started, follow the instructions in the template [README](https://github.com/microsoft/vscode-python-tools-extension-template#readme). There you will learn how to use the [template to create your repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template) and how to install the necessary tools (for example, the [nox](https://nox.thea.codes) task runner) and optional dependencies (testing support).
+
+The [README](https://github.com/microsoft/vscode-python-tools-extension-template#readme) has the most up-to-date instructions and also goes into details on how to customize the extension's `package.json` placeholders (`<pythontool-module>`, `<pythontool-display-name>`, etc.).
+
+## Features of the template
+
+After creating your extension via the template, it will include the following extension contributions. Assume `<pytool-module>` was replaced with `mytool`, and `<pytool-display-name>` with`My Tool`:
+
+1. A command **My Tool: Restart Server** (command ID: `mytool.restart`).
+1. Following settings:
+    * `mytool.logLevel`
+    * `mytool.args`
+    * `mytool.path`
+    * `mytool.importStrategy`
+    * `mytool.interpreter`
+    * `mytool.showNotification`
+1. Following triggers for extension activation:
+    * On Language `python`.
+    * On File with `.py` extension found in the opened workspace.
+    * On Command `mytool.restart`.
+1. Output channel for logging **Output** > **My Tool**.
+
+## Adding features from your tool
+
+The generated `bundled/tool/server.py` file is where you will make most of your changes. `TODO` comments in the file point out the various customization points. Also search for `TODO` comments in other locations in the template, such as other Python and Markdown files. You will want to review the LICENSE file, even if you want to keep it MIT License.
+
+## Examples
+
+There are several example implementations created from the template:
+
+* [Pylint](https://github.com/microsoft/vscode-pylint/tree/main/bundled/tool) - implements linting on file `open`, `save`, and `close`.
```

Comment:
We can also include [flake8](https://github.com/microsoft/vscode-flake8/tree/main/bundled/tool) and [Autopep8](https://github.com/microsoft/vscode-autopep8/tree/main/bundled/tool)

Reply:
Good point! I think it's worth including code actions are also supported.

Reply:
@karthiknadig Is the autopep8 formatter extension available on the Marketplace? I can find the others but not that one? Maybe it is still under development?

Reply:
Added new extensions in last commit as well as call out Code Actions